src/trellis/bundler/utils.py

```
@@ -0,0 +1,36 @@
+    Args:
+        tar: The tarfile to extract
+        dest: The destination directory
+
+    Raises:
+        ValueError: If any member would extract outside the destination
+    """
+    dest = dest.resolve()
+    for member in tar.getmembers():
+        # Reject symlinks and hardlinks to prevent link-based escapes
+        if member.issym() or member.islnk():
+            raise ValueError(f"Tarball contains link entry: {member.name}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)